### DISCUSSION TITLE
Need help to include language functionality into GLSP's vs code client

### DISCUSSION THREAD

**Discussion by sathya-1994** [eclipse-glsp/glsp#384] [Q&A] [answered]
I have configured GLSP's server and now it can support both diagram and language functionality. This server can interact with language client(implemented using LanguageClient of "vscode-languageclient" module) and GLSP's vs code client(to generate diagrams).

https://github.com/eclipse-glsp/glsp-vscode-integration (GLSP's vs code client) supports only diagram functionality. 
How I have to include language functionality into GLSP's vs code client ?

**Comment by planger** [comment]
I'm not sure I completely follow your question.
I assume with language functionality you mean LSP support for textual languages?
The GLSP VSCode integration only supports diagrams by talking to a GLSP server. If you want both, textual and graphical (diagrams) combined in one client, you'd have to extend the GLSP VSCode Integration by including a Monaco. This however is not supported out of the box but needs to be manually implemented.

**Comment by sathya-1994** [comment, marked as answer]
@planger ,

I followed below approach to create a client that supports textual representation and diagrams for textual representation.

Cloned https://github.com/eclipse/sprotty-vscode.git sprotty-vscode project. Instead of using this project's language server, I used mine language server.

Below were the changes related to server option(https://github.com/eclipse/sprotty-vscode/blob/master/example/states/extension/src/states-lsp-extension.ts).
![image](https://user-images.githubusercontent.com/59145404/132514713-55f196b8-c5b5-45b5-9db8-d321884fbb6e.png)

Language functionality(textual representation) is working fine. When I open diagram from text editor, new WebView will be opened but nonfiction/message related to WebView will not be sent to server from language client(Notification related to language functionality are working fine). Also, nonfiction/message will be sent to language client from WebView .

Please help me to resolve this issue. Also, please share any documentation which will be helpful to implement sprotty vscode integration.
.

**Reply by planger** [reply]
Oh, I see you are using sprotty-vscode and not GLSP.
I suggest you should rather ask there at https://github.com/eclipse/sprotty-vscode
While GLSP indeed builds on Sprotty, https://github.com/eclipse/sprotty-vscode and https://github.com/eclipse-glsp/glsp-vscode-integration are still two different projects.
